# Importing libraries and Loading the data 

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [80]:
data = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding='latin1')

In [81]:
data.tail()

Area Code      Area  Item Code           Item  Element Code  \
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit  Y2014  Y2015  \
60938                                    Food      1000 tonnes  42.00  46.00   
60939     Food supply quantity (kg/capita/yr)               kg   3.06   3.33   
60940           Food supply (kcal/capita/day)  kcal/capita/day   3.00   4.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day   0.10   0.11   
60942      Fat supply quantity (g/capita/day)     g/capita/day   0.04   0.05   

       Y2016  Y2017  Y2018  
60938  33.00  19.00  16.00  
60939   2.35   1.33   1.08  
60940   3.00   1.00   1.00  
60941   0.08   0.04   0.04  
60942   0.03   0.02   0.01

In [173]:
data.shape

(60943, 12)

In [83]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [84]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [85]:
data.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

# What is the total number and percentage of missing data in 2014 to 3 decimal places?

### Total Number of missing data in 2014:

In [200]:
tot_mis = data.Y2014.isnull().sum()
tot_mis

1589

**Total Number: 1589**

### Percentage of missing data in 2014:

In [201]:
total = len(data.Y2014)
round(tot_mis/total *100, 3)

2.607

**Percentage: 2.607**

# What is the mean and standard deviation across the whole dataset for the year 2017 to 2 decimal places?

In [209]:
round(data.Y2017.mean(),2),round(data.Y2017.std(),2)

(140.92, 1671.86)

**Answer:**

The mean:140.92

Standard Deviation:1671.86

# Select columns ‘Y2017’ and ‘Area’, Perform a groupby operation on ‘Area’.  Which of these Areas had the 7th lowest sum in 2017?

In [211]:
data.groupby('Area')['Y2017'].sum().sort_values(ascending=True)

Area
Sudan (former)                       0.00
Ethiopia PDR                         0.00
Comoros                             59.84
Seychelles                         442.34
Sao Tome and Principe            12662.63
Cabo Verde                       14650.74
Guinea-Bissau                    19102.77
Lesotho                          21267.96
Botswana                         22101.30
Djibouti                         22729.91
Gambia                           23154.18
Gabon                            27979.64
Liberia                          29342.20
Namibia                          29874.89
Central African Republic         29937.00
Congo                            41181.68
Togo                             49841.88
Mauritius                        51114.83
Eswatini                         54343.33
Sierra Leone                     55311.33
Chad                             71594.68
Rwanda                           73663.69
Zimbabwe                         75919.34
Senegal                      

**Answer: Guinea-Bissau**

# What is the total number of unique countries in the dataset?

In [212]:
data.Area.nunique()

49

**Answer: 49**

# Perform a groupby operation on ‘Element’.  What year has the highest sum of Stock Variation?

In [213]:
stock_var = data[data.Element =='Stock Variation']
stock_var[year_col].sum().sort_values(ascending=False)

Y2014    58749.83
Y2017    54316.91
Y2015    34910.99
Y2016    33140.12
Y2018    20577.91
dtype: float64

**Answer: 2014**

# What is the total Protein supply quantity in Madagascar in 2015?

In [214]:
area_mag = data[data.Area == 'Madagascar']
prot_qt = area_mag[area_mag.Element == 'Protein supply quantity (g/capita/day)']
round(prot_qt.Y2015.sum(), 2)

173.05

**Answer: 173.05**

# Perform a groupby operation on ‘Element’.  What is the total number of the sum of Processing in 2017?

In [215]:
data[data.Element == 'Processing'].Y2017.sum()

292836.0

**Answer: 292836.0**

# Which year had the least correlation with ‘Element Code’?

In [216]:
data.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

**Answer: 2016**

# What is the total sum of Wine produced in 2015 and 2018 respectively?

In [217]:
item_wine= data[data.Item == 'Wine']
round(item_wine.Y2015.sum(),2), round(item_wine.Y2018.sum(),2)

(4251.81, 4039.32)

**Answer:**

- In 2015, sum is 4251.81

- In 2018, sum is 4039.32

# Select columns ‘Y2017’ and ‘Area’, Perform a groupby operation on ‘Area’.  Which of these Areas had the highest sum in 2017?

In [218]:
data.groupby('Area')['Y2017'].sum().sort_values(ascending=False)

Area
Nigeria                        1483268.23
Egypt                           866379.92
South Africa                    517590.54
Ethiopia                        448683.76
Morocco                         388495.36
Ghana                           337599.06
Algeria                         325644.27
United Republic of Tanzania     322616.85
Kenya                           264660.66
Sudan                           239931.92
Cameroon                        232030.43
Angola                          229159.57
Côte d'Ivoire                   224599.01
Uganda                          213950.38
Malawi                          181098.71
Mozambique                      161407.98
Mauritania                      156665.46
Mali                            149928.33
Madagascar                      131197.73
Niger                           126707.58
Benin                           124771.22
Tunisia                         124167.20
Zambia                          103223.77
Burkina Faso                 

**Answer: Nigeria**